capstone/src/main/resources/static/script.js

```diff
@@ -17,25 +17,24 @@ var reviewsExist = false;
  * Retrieves events from server
  */
 function getEvents() {
-  var userType = sessionStorage.getItem("user-type");
-  if (userType == null) {
-    getUserType(false);
-    if (sessionStorage.getItem("user-type") == null) {
-      return;
-    }
-  }
-  var displaySaveButton = userType == "individual";
-  fetch('get-all-events').then(response => response.json()).then((events) => {
-
-    const eventListElement = document.getElementById('events');
-    eventListElement.innerText = ""; // Clear elements in div
-
-    events.forEach((event) => {
-      eventListElement.appendChild(createEventElement(event, displaySaveButton));
-    })
-    // Format time to *** time ago
-    if (reviewsExist){
-      timeago.render(document.querySelectorAll('.timeago'));
+  fetch('user-info').then(response => response.json()).then((data) => {
+    if (data.userType != "null") {
+      var displaySaveButton = data.userType == "individual";
+      fetch('get-all-events').then(response => response.json()).then((events) => {
+
+        const eventListElement = document.getElementById('events');
+        eventListElement.innerText = ""; // Clear elements in div
+
+        events.forEach((event) => {
+          eventListElement.appendChild(createEventElement(event, displaySaveButton));
+        })
+        // Format time to *** time ago
+        if (reviewsExist){
+          timeago.render(document.querySelectorAll('.timeago'));
+        }
+      });
```

Comment:
I'd move this block into a helper function to make the main (is user undefined? show main page. Is it an individual or an organization? fetch all events) logic easier to follow.

Reply:
After I merged with master this piece of code was already condensed (probably by Lian)